### PR TITLE
[docs] add guessing filetype tutorial to API loader docs

### PR DESCRIPTION
**Pull request by anjakefala**

### CODE REVIEW

File: docs/api/loaders.rst [hunk 1]
```diff
@@ -261,6 +261,35 @@ This is a completely functional loader for the ``sas7bdat`` (SAS dataset file) f
...
+
+`Examples of guess_filetype functions <https://github.com/saulpw/visidata/commit/4743f92bb855cf931d896e65845c549ce6027e2f>`
+
+
+::
+
+        @VisiData.api
+        def guess_foo(vd, p):
+            line = foolib.iterfoo(p.open_text())
+            try:
```
Comment (saulpw) [inline docs/api/loaders.rst]: I'm not sure if this `try` is needed anymore.

**Review by saulpw — APPROVED** [review]
This is good; maybe it would help to be explicit about just using the usual `open_text()` and `read()`, but `seek()` may not work.

File: docs/api/loaders.rst (hunk 1 above)
Comment (anjakefala) [reply docs/api/loaders.rst]: I was building off of an earlier `FooSheet` example! I am not attached. I think the `r[0] = "#Foo"` is demonstrated enough (along with the link to actual examples).

**Comment by anjakefala** [comment]
That you could use the standard `read()` to check the first line's contents?

**Comment by saulpw** [comment]
> That you could use the standard `read()` to check the first line's contents?

Yes, or iterate over lines (it's file-like).

**Pull request merged**